examples/gno.land/p/demo/ownable/ownable.gno

```diff
@@ -0,0 +1,57 @@
+package ownable
+
+import (
+	"std"
+)
+
+// Ownable is meant to be used as a top-level object to make your contract ownable OR
+// being embedded in a Gno object to manage per-object ownership.
+type Ownable struct {
+	owner std.Address
+}
+
+func New() *Ownable {
+	return &Ownable{
+		owner: std.GetOrigCaller(),
+	}
+}
+
+// TransferOwnership transfers ownership of the Ownable struct to a new address
+func (o *Ownable) TransferOwnership(newOwner std.Address) error {
+	err := o.CallerIsOwner()
+	if err != nil {
+		return err
+	}
+
+	if !newOwner.IsValid() {
+		return ErrInvalidAddress
+	}
+
+	o.owner = newOwner
+	return nil
+}
+
+// DropOwnership Removes the owner, effectively disabling any owner-related actions
+// Top-level usage: disables all only-owner actions/functions,
+// Embedded usage: behaves like a burn functionality, removing the owner from the struct
+func (o *Ownable) DropOwnership() error {
+	err := o.CallerIsOwner()
+	if err != nil {
+		return err
+	}
+
+	o.owner = ""
+	return nil
+}
+
+// CallerIsOwner checks if the caller of the function is the Realm's owner
+func (o *Ownable) CallerIsOwner() error {
+	if std.GetOrigCaller() == o.owner {
+		return nil
+	}
+	return ErrUnauthorized
+}
+
+func (o *Ownable) GetOwner() std.Address {
```

Comment:
```suggestion
func (o *Ownable) Owner() std.Address {
```

Reply:
I think most packages/realms right now are not thread-safe, and it's a much bigger discussion to be had when actually implementing threads on the execution of the VM. Should be fine for now :)